admiral/pkg/clusters/handler.go

```diff
@@ -103,10 +99,33 @@ func getDestinationRule(host string, locality string, gtpTrafficPolicy *model.Tr
 			dr.TrafficPolicy.LoadBalancer = loadBalancerSettings
 		}
 	}
-	dr.TrafficPolicy.OutlierDetection = outlierDetection
+	dr.TrafficPolicy.OutlierDetection = getOutlierDetection(se, locality, gtpTrafficPolicy)
 	return dr
 }
 
+func getOutlierDetection(se *v1alpha32.ServiceEntry, locality string, gtpTrafficPolicy *model.TrafficPolicy) *v1alpha32.OutlierDetection {
+
+	outlierDetection := &v1alpha32.OutlierDetection{
+		BaseEjectionTime:      &types.Duration{Seconds: 300},
+		ConsecutiveGatewayErrors: &types.UInt32Value{Value: uint32(10)},
+		Interval:              &types.Duration{Seconds: 60},
+	}
+
+	//Scenario 1: Only one endpoint present and is local service (ends in svc.cluster.local) - no outlier detection (optimize this for headless services in future?)
+	if len(se.Endpoints) == 1 && (strings.Contains(se.Endpoints[0].Address, common.DotLocalDomainSuffix) || net.ParseIP(se.Endpoints[0].Address).To4() != nil) {
```

Comment:
Trying to understand the second statement in the condition
```
net.ParseIP...
```
when do we have IPs as part of endpoint addresses?